From 2a797f856eac0006037c108c1cd9cf7020e30c65 Mon Sep 17 00:00:00 2001
From: Martin Bosma <mbosma@victronenergy.com>
Date: Wed, 6 Dec 2023 15:09:08 +0100
Subject: [PATCH 5/6] device: conditionally add position setting

Add position setting only if not provided by the meter.
---
 device.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/device.py b/device.py
index 01f6d9fbc772..a2110d732900 100644
--- a/device.py
+++ b/device.py
@@ -403,6 +403,7 @@ class EnergyMeter(ModbusDevice):
     default_role = 'grid'
     default_instance = 40
     nr_phases = None
+    position = None
 
     def position_setting_changed(self, service, path, value):
         self.dbus['/Position'] = value['Value']
@@ -411,7 +412,7 @@ class EnergyMeter(ModbusDevice):
         super().init_device_settings(dbus)
 
         self.pos_item = None
-        if self.role == 'pvinverter':
+        if self.role == 'pvinverter' and self.position is None:
             self.pos_item = self.settings.addSetting(
                 self.settings_path + '/Position', 0, 0, 2,
                 callback=self.position_setting_changed)
-- 
2.43.0

